### PR TITLE
Detect changes in query lambda types and report rude edits appropriately

### PR DESCRIPTION
**Pull request by tmat**
Fixes RC bug: #1315

### CODE REVIEW

**Comment by tmat** [comment]
@gafter @AlekseyTs - changes documentation comments in the compiler 
@wschae Please review.

File: src/Compilers/VisualBasic/Portable/VisualBasicExtensions.vb [hunk 1]
```diff
@@ -1496,32 +1496,69 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
...
+        '''     <term><see cref="GroupByClauseSyntax"/></term>
+        '''     <description>Returns GroupBy method associated with <see cref="GroupByClauseSyntax"/>.</description>
+        ''' </item>
+        ''' <item>
+        '''     <term><see cref="JoinClauseSyntax"/></term>
+        '''     <description>Returns Join/GroupJoin method associated with <see cref="JoinClauseSyntax"/>.</description>
+        ''' </item>
+        ''' <item>
+        '''     <term><see cref="SelectClauseSyntax"/></term>
+        '''     <description>Returns Join/GroupJoin method associated with <see cref="SelectClauseSyntax"/>, or <see cref="SymbolInfo.None"/> if none is.</description>
```
Comment (wschae) [inline src/Compilers/VisualBasic/Portable/VisualBasicExtensions.vb]: Could it be "Returns Select method..."?

File: src/Compilers/VisualBasic/Portable/VisualBasicExtensions.vb [hunk 2]
```diff
@@ -1537,8 +1574,8 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
         End Function
 
         ''' <summary>
-        ''' Returns Select method associated with ExpressionRangeVariableSyntax within a LetClauseSyntax, if needed.
-        ''' NotNeeded SymbolInfo otherwise.
+        ''' Returns Select method associated with <see cref="ExpressionRangeVariableSyntax"/> within a <see cref="LetClauseSyntax"/>, 
+        ''' or <see cref="SymbolInfo.None"/> otherwise if none is.
```
Comment (wschae) [inline src/Compilers/VisualBasic/Portable/VisualBasicExtensions.vb]: Could we drop "if none is"?

File: src/Features/CSharp/EditAndContinue/CSharpEditAndContinueAnalyzer.cs [hunk 3]
```diff
@@ -902,6 +902,45 @@ internal override SyntaxNode GetLambda(SyntaxNode lambdaBody)
...
+                case SyntaxKind.DescendingOrdering:
+                case SyntaxKind.SelectClause:
+                case SyntaxKind.GroupClause:
+                    var oldSymbolInfo = oldModel.GetSymbolInfo(oldNode, cancellationToken);
+                    var newSymbolInfo = newModel.GetSymbolInfo(newNode, cancellationToken);
+
+                    return MemberSignaturesEquivalent(oldSymbolInfo.Symbol, newSymbolInfo.Symbol);
+
+                default:
+                    return true;
```
Comment (wschae) [inline src/Features/CSharp/EditAndContinue/CSharpEditAndContinueAnalyzer.cs]: What could be this default cases?

Comment (tmat) [inline src/Features/CSharp/EditAndContinue/CSharpEditAndContinueAnalyzer.cs]: Anything that's not a query clause.

**Comment by wschae** [comment]
+1

**Comment by MattGertz** [comment]
Approved for next preview.

**Pull request merged**